packages/pigeon/lib/cpp_generator.dart

```diff
@@ -2,10 +2,14 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io';
+import 'package:path/path.dart' as path;
+
 import 'ast.dart';
 import 'functional.dart';
+import 'generator.dart';
 import 'generator_tools.dart';
-import 'pigeon_lib.dart' show Error;
+import 'pigeon_lib.dart' show Error, PigeonOptions, lineReader, openSink;
```

Comment:
Sonofabitch, late comment.  For architecture questions can you guys touch base first.  I didn't really have an opportunity to react to this.

Reply:
We can revert+retract if this is an issue. I didn't expect putting the generator classes in the generator files to be controversial.

> The idea of the `Generator` was more of an adapter of the public functions in the generator files which were the actual generators.

This is why naming is important; I didn't expect that in a package where the most important conceptual thing are generators that the `Generator` class wasn't intended to represent a generator.

Reply:
Yea, it should have been "_Generator" but was made public to help with testing.  We don't need to retract.  I think separating the adapter and the generator makes everything tidier and easier to maintain but we can do that in a different PR.  I think having a Generator abstract class that people follow along in the generator files makes it a bit easier conceptually instead of having loose functions, so that is an improvement.